app/models/host/discovered.rb

```diff
@@ -247,4 +247,29 @@ def create_notification
   def delete_notification
     ForemanDiscovery::UINotifications::DestroyHost.deliver!(self)
   end
+
+  def notification_recipients_ids
+    org_recipients = find_organization_users if Taxonomy.organizations_enabled
+    org_recipients ||= []
+
+    admins = User.unscoped.only_admin.except_hidden.
+      reorder('').distinct.pluck(:id)
+    (org_recipients + admins).uniq
+  end
+
+  private
+
+  def find_organization_users
+    organization ||= Organization.find_by_name(Setting[:discovery_organization])
+    return [] unless organization.present?
+    organization_users = if organization.ignore_types.include? 'User'
+                           User.unscoped.all
+                         else
+                           organization.users
+                         end
+
+    organization_users.find_all do |user|
+      user.can? :create_hosts
+    end.pluck(:id)
+  end
```

Comment:
This assumes that all hosts are always discovered in `Setting[:discovery_organization]` but that's not the case. We have pretty complex (and weird) mechanism of finding taxonomy described here: https://theforeman.org/plugins/foreman_discovery/9.1/index.html#3.1.3OrganizationsandLocations

I guess you need to use discovered host's organization instead of setting here. This needs to be trigerred tho after host import (we set taxonomy in `populate_fields_from_facts`).

Reply:
I was using `host.organization` before, but yeah it was empty as this happens prior to `populate_fields_from_facts`. I'll change that.